packages/features/ee/teams/lib/removeMember.ts

```diff
@@ -101,16 +113,91 @@ const removeMember = async ({
       }),
       ProfileRepository.delete({
         userId: membership.userId,
-        organizationId: orgInfo.id,
+        organizationId: team.id,
       }),
     ]);
+
+    // delete all workflowReminders of the removed team member that come from org workflows
+    const workflowRemindersToDelete = await prisma.workflowReminder.findMany({
+      where: {
+        workflowStep: {
+          workflow: {
+            teamId: team.id,
+          },
+        },
+        booking: {
+          eventType: {
+            userId: memberId,
+          },
+        },
+      },
+      select: {
+        id: true,
+        referenceId: true,
+        method: true,
+      },
+    });
+
+    deleteAllWorkflowReminders(workflowRemindersToDelete);
```

Comment:
We should abstract these lines of code out in smaller functions with good names to convey the right meaning and make them unit testable as they can make or break things.

Reply:
Also, as a followup maybe we should move these queries to workflow repository

Reply:
> Also, as a followup maybe we should move these queries to workflow repository

I agree with that, I will create a follow up issue 🙏